client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,135 @@
+    const onSoftDeleteRequest = (index: number) => {
+        const requestsCopy = requests.slice();
+        requestsCopy.splice(index, 1);
+        const id = requestsCopy[index]._id;
+        props.onChangeNumRequests!(requestsCopy.length);
+        setRequests(requestsCopy);
+        mutateDeleteRequest({variables: {id: id}});
+    }
+
+    const onFulfilledRequest = (index: number) => {
+        requests[index].fulfilled = !requests[index].fulfilled;
+        orderRequests(requests);
+        const id = requests[index]._id;
+        const requestId = requests[index].requestId;
+        const fulfilled = requests[index].fulfilled;
+        mutateRequest({variables:{request: {id, requestId, fulfilled}}});
+    }
+
+
+    return (
+        <div className="client-request-table">
+            { requestSelectedForEditing && <RequestForm onSubmitComplete={() => { window.location.reload() }} handleClose={() => setRequestSelectedForEditing("")} operation="edit" requestId={requestSelectedForEditing} /> }
+            {requests.length !== 0 &&
+                <Table responsive className="request-table"> 
+                <thead> 
+                <tr>
+                    {headingList.map((heading, index) => (
+                    <th key={index} className="request-table th">{heading}</th>
+                    ))}
+                </tr>
+                </thead>
+                <tbody> 
+                    {requests.map((request, index)=> ( 
+                        request.deleted === false ?
+                        <tr key={request._id}>
+                            <td>
+                                <div><Form.Check type="checkbox" onClick={() => onFulfilledRequest(index)} defaultChecked={request.fulfilled}/></div>
+                            </td>
+                            {request?.requestType?.requestGroup != null ?
+                                <td style={request.fulfilled ? {opacity: 0.2}:undefined}><div className="row-text-style">{request.requestType.requestGroup.name}</div></td>
+                                : <td>N/A</td>
+                            }
+                            {request.requestType != null ?
+                                <td style={requests[index].fulfilled ? {opacity: 0.2}:undefined}><div className="row-text-style">{request.requestType.name}</div></td>
+                                : <td>N/A</td>
+                            }
+                            <td style={request.fulfilled ? {opacity: 0.2}:undefined}>{request.quantity}</td>
+                            <td style={request.fulfilled ? {opacity: 0.2}:undefined}>{
+                                moment(request.dateCreated, "x").format('MMMM DD, YYYY')
+                            }</td>
+                            <td><div className="btn-cont">
+                                <td><a className="request-table edit" onClick={() => { if(request._id) { setRequestSelectedForEditing(request._id) }}}><i className="bi bi-pencil"></i></a></td>
```

Comment:
Each request will have an id associated with it so I'm not too sure if you need to have an if(request._id), 
Maybe
```
...onClick={() => { if(request)
```

Reply:
Hmm I think I run into an error since all those fields being optional though, so it could be undefined or a string, and so the `setRequestSelectedForEditing` gets mad :/